crates/runtime/src/table.rs

```diff
@@ -137,10 +173,37 @@ impl Table {
 
     /// Return a `VMTableDefinition` for exposing the table to compiled wasm code.
     pub fn vmtable(&self) -> VMTableDefinition {
-        let mut vec = self.vec.borrow_mut();
-        VMTableDefinition {
-            base: vec.as_mut_ptr() as *mut u8,
-            current_elements: vec.len().try_into().unwrap(),
+        match &*self.elements.borrow() {
+            TableElements::FuncRefs(x) => VMTableDefinition {
+                base: x.as_ptr() as *const u8 as *mut u8,
+                current_elements: x.len().try_into().unwrap(),
+            },
+            TableElements::ExternRefs(x) => VMTableDefinition {
+                base: x.as_ptr() as *const u8 as *mut u8,
```

Comment:
We are exposing `Option<VMExternRef>` elements to the CL or other low level compiler. It is hard to guess what type of binary representation we will be dealing with here. For comparison, I see `#[repr(C)]` near `VMCallerCheckedAnyfunc`.

Reply:
`VMExternRef` is `repr(transparent)` around a `NonNull` which is `repr(transparent)` around a pointer. Additionally, `Option<NonNull<...>>` is guaranteed to have the same representation as a pointer. So we should be good as far as binary representation goes.